typescript/bedrock-kb-agent/lib/constructs/cloudfront-s3-website-construct.ts

```diff
@@ -0,0 +1,220 @@
+   */
+  readonly userPoolId: string;
+
+  /**
+   * The Cognito AppClientId to authenticate users in the front-end
+   */
+  readonly appClientId: string;
+
+  /**
+   * The Cognito IdentityPoolId to authenticate users in the front-end
+   */
+  readonly identityPoolId: string;
+
+  /**
+   * The name of the bucket to use as storage using Amplify SDK if required
+   */
+  readonly storageBucketName?: string;
+
+  /**
+   * If you use Amplify SDK to access an API serving under /api path set this boolean to true
+   */
+  readonly withApi?: boolean;
+}
+
+const defaultProps: Partial<CloudFrontS3WebSiteConstructProps> = {};
+
+/**
+ * Deploys a CloudFront Distribution pointing to an S3 bucket containing the deployed web application {webSiteBuildPath}.
+ * Creates:
+ * - S3 bucket
+ * - CloudFrontDistribution
+ * - OriginAccessIdentity
+ *
+ * On redeployment, will automatically invalidate the CloudFront distribution cache
+ */
+export class CloudFrontS3WebSiteConstruct extends Construct {
+  /**
+   * The origin access identity used to access the S3 website
+   */
+  public originAccessIdentity: cdk.aws_cloudfront.OriginAccessIdentity;
+
+  /**
+   * The cloud front distribution to attach additional behaviors like `/api`
+   */
+  public cloudFrontDistribution: cdk.aws_cloudfront.Distribution;
+
+  /**
+   * The name of the bucket where the frontend assets are stored
+   */
+  public siteBucket: cdk.aws_s3.Bucket;
+
+  constructor(
+    parent: Construct,
+    name: string,
+    props: CloudFrontS3WebSiteConstructProps,
+  ) {
+    super(parent, name);
+
+    props = { ...defaultProps, ...props };
+    const stack = cdk.Stack.of(this);
+    const region = stack.region;
+
+    // When using Distribution, do not set the s3 bucket website documents
+    // if these are set then the distribution origin is configured for HTTP communication with the
+    // s3 bucket and won't configure the cloudformation correctly.
+    this.siteBucket = new cdk.aws_s3.Bucket(this, "WebApp", {
+      encryption: cdk.aws_s3.BucketEncryption.S3_MANAGED,
+      autoDeleteObjects: true,
+      blockPublicAccess: cdk.aws_s3.BlockPublicAccess.BLOCK_ALL,
+      removalPolicy: cdk.RemovalPolicy.DESTROY,
+      enforceSSL: true,
+    });
+
+    this.siteBucket.addToResourcePolicy(
+      new cdk.aws_iam.PolicyStatement({
+        sid: "EnforceTLS",
+        effect: cdk.aws_iam.Effect.DENY,
+        principals: [new cdk.aws_iam.AnyPrincipal()],
+        actions: ["s3:*"],
+        resources: [
+          this.siteBucket.bucketArn,
+          this.siteBucket.bucketArn + "/*",
+        ],
+        conditions: { Bool: { "aws:SecureTransport": "false" } },
+      }),
+    );
+
+    const originAccessIdentity = new cdk.aws_cloudfront.OriginAccessIdentity(
+      this,
+      "OriginAccessIdentity",
+    );
+    this.siteBucket.grantRead(originAccessIdentity);
+
+    const s3origin = new cdk.aws_cloudfront_origins.S3Origin(this.siteBucket, {
```

Comment:
S3Origin has been deprecated can you update to S3OriginBucket.

Also your tests are failing please fix.